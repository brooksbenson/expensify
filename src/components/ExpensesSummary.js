import React from 'react';
import numeral from 'numeral';

export default ({expensesTotal, expensesCount}) => {
  const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
  const formattedExpenseTotal = numeral(expensesTotal / 100).format('$0,0.00'); 
  return (
    <div> 
      <h1>Viewing {expensesCount} {expenseWord} totalling {formattedExpenseTotal}</h1>
    </div>
  );
};