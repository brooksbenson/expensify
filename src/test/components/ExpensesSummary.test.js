import React from 'react';
import { shallow } from 'enzyme';
import ExpensesSummary from '../../components/ExpensesSummary';

test('should render ExpensesSummary correctly with no expenses', () => {
  const expensesTotal = 0;
  const expensesCount = 0;
  const wrapper = shallow(
    <ExpensesSummary 
      expensesTotal={expensesTotal}
      expensesCount={expensesCount}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with one expense', () => {
  const expensesTotal = 100;
  const expensesCount = 1;
  const wrapper = shallow(
    <ExpensesSummary 
      expensesTotal={expensesTotal}
      expensesCount={expensesCount}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpensesSummary correctly with multiple expenses', () => {
  const expensesTotal = 300;
  const expensesCount = 3;
  const wrapper = shallow(
    <ExpensesSummary 
      expensesTotal={expensesTotal}
      expensesCount={expensesCount}
    />
  );
  expect(wrapper).toMatchSnapshot();
});