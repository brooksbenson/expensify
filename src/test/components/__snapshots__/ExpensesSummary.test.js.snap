// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $3.00
  </h1>
</div>
`;

exports[`should render ExpensesSummary correctly with no expenses 1`] = `
<div>
  <h1>
    Viewing 
    0
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;

exports[`should render ExpensesSummary correctly with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.00
  </h1>
</div>
`;
